### ISSUE TITLE
convex_hull_image only works for C-contiguous arrays

### ISSUE THREAD

**Issue by Dimchord** [scikit-image/scikit-image#3758]
## Description
The convex_hull_image function only works for C-contiguous arrays, not for Fortran-contiguous or non-contiguous arrays.

## Way to reproduce
```python
from skimage.morphology import convex_hull_image
convex_hull_image(np.ones((2,2), order='F'))
```
Output:
```python
Traceback (most recent call last):

  File "<ipython-input-47-960dea2150d3>", line 1, in <module>
    convex_hull_image(mask == 1)

  File "C:\Users\robk\Anaconda3\lib\site-packages\skimage\morphology\convex_hull.py", line 59, in convex_hull_image
    coords = possible_hull(image.astype(np.uint8))

  File "skimage\morphology\_convex_hull.pyx", line 10, in skimage.morphology._convex_hull.possible_hull

  File "stringsource", line 653, in View.MemoryView.memoryview_cwrapper

  File "stringsource", line 348, in View.MemoryView.memoryview.__cinit__

ValueError: ndarray is not C-contiguous
```

Should be easy to fix by changing the offending line in convex_hull.py to

```python
coords = possible_hull(np.ascontiguousarray(image.astype(np.uint8)))
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
3.6.5 |Anaconda custom (64-bit)| (default, Mar 29 2018, 13:32:41) [MSC v.1900 64 bit (AMD64)]
Windows-10-10.0.17763-SP0
scikit-image version: 0.14.1
numpy version: 1.14.3
```

**Comment by hmaarrfk** [comment]
Do you want to submit the patch with the little test?

**Comment by Dimchord** [comment]
Something like this?

[patch_convex_hull.txt](https://github.com/scikit-image/scikit-image/files/2883840/patch_convex_hull.txt)

**Comment by hmaarrfk** [comment]
looks good :)

**Comment by jni** [comment]
@Dimchord very nice! (And thanks for noticing!) Would you like to submit a pull request (see our [contributing guide](http://scikit-image.org/docs/dev/contribute.html)) or should we just pilfer your diff? =)

**Comment by Dimchord** [comment]
I'll have a go at a pull request later today.

**Comment by lagru** [comment]
Closing this as the fix in #3762 seems to be merged.
